src/parser/internal/identifiers.rs

```diff
@@ -96,16 +96,56 @@ pub fn ident_maybe_reserved(state: &mut State) -> ParseResult<Identifier> {
     }
 }
 
+pub fn ident_maybe_soft_reserved(state: &mut State) -> ParseResult<Identifier> {
+    match state.current.kind {
+        _ if is_soft_reserved_ident(&state.current.kind) => {
+            let name = state.current.kind.to_string().into();
+
+            let start = state.current.span;
+            state.next();
+            let end = state.current.span;
+
+            Ok(Identifier { start, name, end })
+        }
+        _ => ident(state),
+    }
+}
+
+pub fn is_ident_maybe_soft_reserved(kind: &TokenKind) -> bool {
+    if let TokenKind::Identifier(_) = kind {
+        return true;
+    }
+
+    is_soft_reserved_ident(kind)
+}
+
+pub fn is_ident_maybe_reserved(kind: &TokenKind) -> bool {
+    if let TokenKind::Identifier(_) = kind {
+        return true;
+    }
+
+    is_reserved_ident(kind)
+}
+
+pub fn is_soft_reserved_ident(kind: &TokenKind) -> bool {
+    matches!(kind, |TokenKind::Parent| TokenKind::Self_
+        | TokenKind::True
+        | TokenKind::False
+        | TokenKind::Null
+        | TokenKind::Enum
+        | TokenKind::From
+        | TokenKind::Readonly)
+}
```

Comment:
we could actually allow much more keywords to be used for function names/calls ( class, interface, trait, extends, implements, private, public, protected, static, or, xor, and, endif, endwhile, endswitch, .. etc ), but PHP doesn't, so 🤷‍♂️